tests/utils.py

```diff
@@ -42,7 +71,72 @@ def netloc(self) -> Tuple[bytes, int]:
 
     @property
     def host_header(self) -> Tuple[bytes, bytes]:
-        return (b"host", self._host.encode("utf-8"))
+        return (b"host", self._host.encode("ascii"))
+
+
+class HypercornServer(Server):  # pragma: no cover  # Python 3.7+ only
```

Comment:
For the sake of simplicity I'm `nocover`-ing all these classes that are only used either on 3.6 (where `hypercorn` isn't available) or on 3.7+ (where `hypercorn` is available), but I'm also aware that something like [conditional coverage](https://sobolevn.me/2020/02/conditional-coverage) could be used to define rules allowing us to only skip coverage on specific versions. For now, keeping it simple. :-)